packages/nimble-components/src/wafer-map/tests/experimental-computations.spec.ts

```diff
@@ -0,0 +1,187 @@
+            right: 4,
+            bottom: 4,
+            left: 4
+        };
+        beforeEach(() => {
+            const waferMock = getWaferMapMockComputationsExperimental(
+                getWaferMapDiesTable(),
+                WaferMapOriginLocation.topLeft,
+                100,
+                100
+            );
+            computationsModule = new Computations(waferMock as WaferMap);
+            computationsModule.updateContainerDimensions();
+        });
+
+        it('should have expected square container', () => {
+            expect(computationsModule.containerDimensions).toEqual({
+                width: 92,
+                height: 92
+            });
+        });
+
+        it('should have expected die size', () => {
+            const computedDimensions = {
+                width: Math.ceil(computationsModule.dieDimensions.width),
+                height: Math.ceil(computationsModule.dieDimensions.height)
+            };
+            expect(computedDimensions).toEqual({
+                width: 19,
+                height: 16
+            });
+        });
+
+        it('should have expected margin', () => {
+            expect(computationsModule.margin).toEqual(expectedMargin);
+        });
+
+        it('should have horizontal domain containing min and max column indexes', () => {
+            expect(computationsModule.horizontalScale.domain()).toEqual([2, 7]);
+        });
+        it('should have vertical domain containing min and max  row indexes, ', () => {
+            expect(computationsModule.verticalScale.domain()).toEqual([1, 7]);
+        });
+    });
+
+    describe('with rectangular canvas', () => {
+        beforeEach(() => {
+            const waferMock = getWaferMapMockComputationsExperimental(
+                getWaferMapDiesTable(),
+                WaferMapOriginLocation.topLeft,
+                200,
+                100
+            );
+            computationsModule = new Computations(waferMock as WaferMap);
+            computationsModule.updateContainerDimensions();
+        });
+
+        it('should have adjusted square container', () => {
+            expect(computationsModule.containerDimensions).toEqual({
+                width: 92,
+                height: 92
+            });
+        });
+
+        it('should have adjusted die size', () => {
+            const computedDimensions = {
+                width: Math.ceil(computationsModule.dieDimensions.width),
+                height: Math.ceil(computationsModule.dieDimensions.height)
+            };
+            expect(computedDimensions).toEqual({
+                width: 19,
+                height: 16
+            });
+        });
+
+        it('should have adjusted margin', () => {
+            expect(computationsModule.margin).toEqual({
+                top: 4,
+                right: 54,
+                bottom: 4,
+                left: 54
+            });
+        });
+    });
+
+    describe('with top left originLocation', () => {
```

Comment:
It feels like groups of these tests could be handled as a `parameterizeSpec`. For this `describe` and the following three, the only differences are the provided orientation, and the provided expected results for the horizontal and vertical ranges. These should all be able to be captured as `parameterizeSpec` parameters.

Consider other `parameterizeSpec` cases as well for providing a variety of input parameters and expected results.